src/SipHash.chpl

```diff
@@ -49,32 +49,28 @@ module SipHash {
     return c;
   }
   
-  proc sipHash64(msg: [] uint(8), k: [?kD] uint(8)): uint(64) throws {
-    var res = computeSipHash(msg, k, 8);
+  proc sipHash64(msg: [] uint(8), D): uint(64) throws {
+    var res = computeSipHash(msg, D, 8);
     return res[1];
   }
 
-  proc sipHash128(msg: [] uint(8), k: [?kD] uint(8)): 2*uint(64) throws {
-    return computeSipHash(msg, k, 16);
+  proc sipHash128(msg: [] uint(8), D): 2*uint(64) throws {
+    return computeSipHash(msg, D, 16);
   }
   
-  private proc computeSipHash(msg: [?D] uint(8), k: [?kD] uint(8), outlen: int) throws {
+  private proc computeSipHash(msg: [] uint(8), D, param outlen: int) throws {
```

Comment:
I didn't mean to commit making outlen a `param` (though it was something I was playing with since it would eliminate some runtime branches.) I might leave that change out for now and let me experiment with that some more.

Reply:
@reuster986 do you want to make this change?

Reply:
@ronawho sorry for stealing your work-in-progress code, I just had to run with something quickly. I don't mind leaving outlen as a `param` in this PR, since it works and should be faster (right?). If in the course of your experimentation you decide to change it back, we can just apply your PR on top of this one.

The other change that's in this line is that I took out the hash key, `k`, as an argument and instead hard-coded a value for it inside the function. The `k` argument was incurring some extra communication in multi-locale setting that really messed with the scaling, and we don't really need a keyed hash here anyway.

Reply:
> sorry for stealing your work-in-progress code

Not a problem, glad that was enough of an improvement to be helpful

>  I don't mind leaving outlen as a param in this PR...

Sounds good. I tend to be pretty methodical (and usually pretty slow) with performance changes just so I can keep track of what changes improved what and by how much, but you're right that it should be strictly better.